src/transforms/compound.rs

```diff
@@ -0,0 +1,93 @@
+use crate::{
+    config::{
+        DataType, ExpandType, GenerateConfig, GlobalOptions, TransformConfig, TransformDescription,
+    },
+    transforms::Transform,
+};
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct CompoundConfig {
+    nested: IndexMap<String, Box<dyn TransformConfig>>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<CompoundConfig>("compound")
+}
+
+impl GenerateConfig for CompoundConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            nested: IndexMap::new(),
+        })
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "compound")]
+impl TransformConfig for CompoundConfig {
+    async fn build(&self, _globals: &GlobalOptions) -> crate::Result<Transform> {
+        Err("this transform must be expanded".into())
+    }
+
+    fn expand(
+        &mut self,
+    ) -> crate::Result<Option<(IndexMap<String, Box<dyn TransformConfig>>, ExpandType)>> {
+        if !self.nested.is_empty() {
+            Ok(Some((self.nested.clone(), ExpandType::Serial)))
+        } else {
+            Err("must specify at least one transform".into())
```

Comment:
Is this a strict requirement because of the implementation?

I realize it isn't functionally useful, but I could see allowing users to have an empty `compound` transform that just passes through events. This could be useful if, for example, they wanted to temporarily comment out all of the serial transforms.

Reply:
Bouncing on the other comment, I think this could be addressed by having a top level passthrough transform. Ideally this noop transform would bear the name of the `compound` transform as it is defined in the config, it would also be usable to generate top level metrics as you suggested above, WDYT ?

Reply:
👍 yeah, a noop transform seems like it would help for that case. We can follow up on it later though; I think it'll be rare that a user would hit this.